### PR TITLE
Enhance bot api code readability and adding some Kdoc

### PR DESCRIPTION
**Pull request by djowalker**
Some improvements made about Kdoc and code organisation in tock bot api.

### CODE REVIEW

File: bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt [hunk 1]
```diff
@@ -23,34 +23,31 @@ private fun defaultUnknownStory() = unknownStory { end("Sorry I didn't understan
...
-     */
     handler: (ClientBus).() -> Unit
 ) = ClientStoryDefinition(Intent.unknown, handler = newStoryHandler(handler))
 
 /**
- * Creates a new bot.
+ * Creates a new bot [ClientBotDefinition] in bot Api mode
+ * @param apiKey the api key in studio for the bot configuration
+ * @param stories List of stories [ClientStoryDefinition] supported by the bot.
+ * @param unknownStory the unknown story triggered when unknown intent
```
Comment (Fabilin) [inline bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt:36]:
```suggestion
 * @param unknownStory the story to trigger when the `unknown` intent is selected
```

File: bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt [hunk 2]
```diff
@@ -23,34 +23,31 @@ private fun defaultUnknownStory() = unknownStory { end("Sorry I didn't understan
...
-    /**
-     * List of stories supported by the bot.
-     */
     stories: List<ClientStoryDefinition>,
     unknownStory: ClientStoryDefinition = defaultUnknownStory()
 ): ClientBotDefinition = ClientBotDefinition(apiKey, stories, unknownStory)
 
 /**
- * Creates a new bot.
+ * Creates a new bot [ClientBotDefinition] in bot Api mode
```
Comment (Fabilin) [inline bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt:45]:
```suggestion
 * Creates a new [bot][ClientBotDefinition] in bot Api mode
```

File: bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt [hunk 3]
```diff
@@ -23,34 +23,31 @@ private fun defaultUnknownStory() = unknownStory { end("Sorry I didn't understan
...
+ * @param apiKey the api key in studio for the bot configuration
```
Comment (Fabilin) [inline bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt:46]:
```suggestion
 * @param apiKey the api key for the bot configuration, as found in  Tock Studio
```

File: bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt [hunk 4]
```diff
@@ -23,34 +23,31 @@ private fun defaultUnknownStory() = unknownStory { end("Sorry I didn't understan
...
+ * @param stories List of stories [ClientStoryDefinition] supported by the bot.
```
Comment (Fabilin) [inline bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt:47]:
```suggestion
 * @param stories List of [stories][ClientStoryDefinition] supported by the bot.
```

File: bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt [hunk 5]
```diff
@@ -60,20 +57,20 @@ fun newBot(
     )
 
 /**
- * Creates a new story.
+ * Creates a new story [ClientStoryDefinition] in bot Api mode
```
Comment (Fabilin) [inline bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt:60]:
```suggestion
 * Creates a new [story][ClientStoryDefinition] in bot Api mode
```

File: bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt [hunk 6]
```diff
@@ -60,20 +57,20 @@ fun newBot(
...
+ * @param mainIntent [String] The main intent name.
```
Comment (Fabilin) [inline bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt:61]: Is specifying the type really necessary here?

File: bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt [hunk 7]
```diff
@@ -60,20 +57,20 @@ fun newBot(
...
+ * @param otherStarterIntents other intents that triggers the story
+ * @param secondaryIntents other intents available in the story scope when it is triggered
+ * @param steps List of steps stories [ClientStep]
```
Comment (Fabilin) [inline bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt:64]:
```suggestion
 * @param steps List of [story steps][ClientStep]
```

**Review by Fabilin — COMMENTED** [review]
Just a few nitpicks - thought kdoc looked nicer with proper links

File: bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt (hunk 6 above)
Comment (djowalker) [reply bot/api/client/src/main/kotlin/ClientDefinitionBuilders.kt:61]: Not sure but there is the one with IntentAware below.

**Review by Fabilin — APPROVED** [review]

**Review by vsct-jburet — APPROVED** [review]

**Comment by vsct-jburet** [comment]
Thank you !

**Pull request merged**